src/PointwiseFunctions/AnalyticSolutions/Elasticity/BentBeam.hpp

```diff
@@ -0,0 +1,150 @@
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Elasticity {
+namespace Solutions {
+/*!
+ * \brief A state of pure bending of an elastic beam in 2D
+ *
+ * \details This solution describes a 2D slice through an elastic beam of length
+ * \f$L\f$ and height \f$H\f$ that is subject to a bending moment \f$M=\int
+ * T^{xx}y\mathrm{d}y\f$ (see e.g. \cite ThorneBlandford2017, Eq. 11.41c for a
+ * bending moment in 1D). The beam material is characterized by an isotropic and
+ * homogeneous constitutive relation \f$Y^{ijkl}\f$ in the plane-stress
+ * approximation (see
+ * `Elasticity::ConstitutiveRelations::IsotropicHomogeneous`). In this scenario,
+ * no body-forces \f$f_\mathrm{ext}^j\f$ act on the material, so the
+ * \ref Elasticity equations reduce to \f$\nabla_i T^{ij}=0\f$, but the bending
+ * moment \f$M\f$ generates the stress
+ * \f[
+ * T^{xx} = \frac{12 M}{H^3} y \\
+ * T^{xy} = 0 = T^{yy} \text{.}
+ * \f]
+ * By fixing the rigid-body motions to
+ * \f[
+ * u^x(0,y)=0 \quad \text{and} \quad u^y(\pm \frac{L}{2},0)=0
+ * \f]
+ * we find that this stress is produced by the displacement field
+ * \f{align}
+ * u^x&=-\frac{12 M}{EH^3}xy
+ * u^y&=\frac{6 M}{EH^3}(x^2+\nu y^2-\frac{L^2}{4})
+ * \f}
+ * in terms of the Young's modulus \f$E\f$ and the Poisson ration \f$\nu\f$ of
+ * the material.
+ */
+class BentBeam {
+ public:
+  using constitutive_relation_type =
+      Elasticity::ConstitutiveRelations::IsotropicHomogeneous<2>;
+
+  struct Length {
+    using type = double;
+    static constexpr OptionString help{"The beam length"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct Height {
+    using type = double;
+    static constexpr OptionString help{"The beam height"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+  struct BendingMoment {
+    using type = double;
+    static constexpr OptionString help{
+        "The bending moment applied to the beam"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  using options = tmpl::list<
+      Length, Height, BendingMoment,
+      Elasticity::Tags::ConstitutiveRelation<constitutive_relation_type>>;
+  static constexpr OptionString help{
+      "A 2D slice through an elastic beam which is subject to a bending "
+      "moment. The bending moment is applied along the length of the beam, "
+      "i.e. the x-axis, so that the beam's left and right ends are bent "
+      "towards the positive y-axis. It is measured in units of force."};
+
+  BentBeam() = default;
+  BentBeam(const BentBeam&) noexcept = delete;
+  BentBeam& operator=(const BentBeam&) noexcept = delete;
+  BentBeam(BentBeam&&) noexcept = default;
+  BentBeam& operator=(BentBeam&&) noexcept = default;
+  ~BentBeam() noexcept = default;
+
+  BentBeam(double length, double height, double bending_moment,
+           constitutive_relation_type constitutive_relation) noexcept;
+
+  double length() const noexcept { return length_; }
```

Comment:
why do these (except maybe the constitutive relation) need to be exposed? None of our evolution solutions expose any member variables except the equation of state.

Reply:
I can't remember why I did that, remove it for now. Kept the `constitutive_relation` accessor because that's probably needed.